Source/Scene/Scene.js

```diff
@@ -3043,6 +3048,16 @@ define([
      * @exception {DeveloperError} Picking from the depth buffer is not supported. Check pickPositionSupported.
      */
     Scene.prototype.pickPosition = function(windowPosition, result) {
+        var cacheKey;
+
+        if(defined(windowPosition)){
+            cacheKey = windowPosition.x + '-' + windowPosition.y;
+        }
+
+        if(defined(windowPosition) && this._pickPositionCache.hasOwnProperty(cacheKey)){
+            return Cartesian2.clone(this._pickPositionCache[cacheKey], result);
+        }
```

Comment:
When this code is moved to `pickPositionWorldCoordinates` the `defined(windowPosition)` checks here and above won't be needed.